src/user-schema.js

```diff
@@ -0,0 +1,87 @@
+const dynamoose = require('dynamoose')
+const moment = require('moment')
+const _chunk = require('lodash.chunk')
+
+const batchGetLimit = 25
+
+dynamoose.setDefaults({
+  create: false,
+  waitForActiveTimeout: 5000
+})
+
+const Schema = dynamoose.Schema
+
+const expiryOffset = {
+  value: 2,
+  unit: 'hours'
+}
+
+const calculateExpiry = (offset) => {
+  return moment().add(offset.value, offset.unit).unix()
+}
+
+const userSchema = new Schema({
+  primary_id: {
+    type: String,
+    hashKey: true
+  },
+  loan_ids: [String],
+  request_ids: [String],
+  expiry_date: {
+    type: Number,
+    default: (model) => calculateExpiry(expiryOffset)
+  }
+}, {})
+
+userSchema.methods = {
+  getData: function (model, source, tableKey) {
+    const batches = _chunk(source, batchGetLimit)
+    let promises = []
+    let retrievedData = []
+
+    batches.forEach((batch) => {
+      const batchKeys = batch.map((id) => {
+        let getMethodKey = {}
+        getMethodKey[tableKey] = id
+        return getMethodKey
+      })
+      promises.push(model.batchGet(batchKeys)
+        .then(data => {
+          retrievedData = retrievedData.concat(data)
+        }))
+    })
+
+    return Promise.all(promises).then(() => retrievedData)
+  },
+
+  getLoanData: function (loanModel) {
+    return this.getData(loanModel, this.loan_ids, 'loan_id')
```

Comment:
Are you not creating instances of loan objects on the class in an array just storing the Ids ?

Reply:
I've added a test in https://github.com/lulibrary/LAG-Alma-Utils/pull/17/commits/900740b266706473d68ade946dc4666acea121f0 which confirms that the result of getData (and hence getLoanData/getRequestData) is an array loan/request objects